h2/src/main/org/h2/value/ValueArray.java

```diff
@@ -61,6 +61,39 @@ public static ValueArray getEmpty() {
         return (ValueArray) EMPTY;
     }
 
+    /**
+     * Create an array value by concatenating this and the passed value.
+     *
+     * @param o an array to add.
+     * @return a new array.
+     * @throws IllegalArgumentException if {@link #getComponentType() component types} aren't equal.
+     */
+    public Value concatenate(ValueArray o) {
+        if (!this.getComponentType().equals(o.getComponentType()))
+            throw new IllegalArgumentException(
+                    "Expected component type " + this.getComponentType() + " but got " + o.getComponentType());
+        final Value[] res = Arrays.copyOf(this.values, this.values.length + o.values.length);
+        System.arraycopy(o.values, 0, res, this.values.length, o.values.length);
+        return ValueArray.get(this.getComponentType(), res);
+    }
+
+
+    /**
+     * Create an array value by adding a new item at the end.
+     *
+     * @param item the value to add.
+     * @return a new array.
+     * @throws IllegalArgumentException if <code>item</code> is of the wrong {@link #getComponentType() type}.
+     */
+    public Value append(Value item) {
+        if (item != ValueNull.INSTANCE && !this.getComponentType().isInstance(item.getObject()))
```

Comment:
It's better to avoid `Value.getObject()` here if component type is not specified (it usually should not be specified, H2 doesn't really have typed arrays). This method may be slow.